plugins/src/opengrok/auth/plugin/ldap/LdapFacade.java

```diff
@@ -292,22 +320,26 @@ private SearchControls prepareSearchControls() {
         if (errorTimestamp > 0 && errorTimestamp + interval > System.currentTimeMillis()) {
             if (!reported) {
                 reported = true;
-                LOGGER.log(Level.SEVERE, "Server pool is still broken");
+                LOGGER.log(Level.SEVERE, "LDAP server pool is still broken");
             }
-            return null;
+            throw new LdapError("LDAP server pool is still broken");
```

Comment:
why don't you throw a checked exception and convert it to unchecked in the plugin itself?